src/Routing/Types.purs

```diff
@@ -0,0 +1,6 @@
+module Routing.Types where
+
+import qualified Data.StrMap as M
+
+data RoutePart = Path String | Query (M.StrMap String)
+type Route = [RoutePart]
```

Comment:
I understand. But I believe that ability of matching routes with many query parts is useful